src/kronecker.jl

```diff
@@ -0,0 +1,258 @@
+    if apply
+        A_mul_B!(y, kron(prod(As), prod(Bs)), x)
+    else
+        A_mul_B!(y, CompositeMap{T}(map(LinearMap, L.maps)), x)
+    end
+end
+
+function _kronmul!(y, B, X, At, T)
+    na, ma = size(At)
+    mb, nb = size(B)
+    v = zeros(T, ma)
+    Ty = eltype(y)
+    temp1 = Array{Ty}(undef, na)
+    temp2 = Array{Ty}(undef, nb)
+    @views @inbounds for i in 1:ma
+        v[i] = one(T)
+        A_mul_B!(temp1, At, v)
+        A_mul_B!(temp2, X, temp1)
+        A_mul_B!(y[((i-1)*mb+1):i*mb], B, temp2)
+        v[i] = zero(T)
+    end
+    return y
+end
+function _kronmul!(y, B::Union{MatrixMap,UniformScalingMap}, X, At::Union{MatrixMap,UniformScalingMap}, T)
+    na, ma = size(At)
+    mb, nb = size(B)
+    if (nb + ma) * na < (ma + mb) * nb
+        mul!(reshape(y, (mb, ma)), B, convert(Matrix, X*At))
+    else
+        mul!(reshape(y, (mb, ma)), convert(Matrix, B*X), At isa MatrixMap ? At.lmap : At.λ)
+    end
+    return y
+end
+
+LinearMaps.At_mul_B!(y::AbstractVector, A::KroneckerMap, x::AbstractVector) = A_mul_B!(y, transpose(A), x)
+
+LinearMaps.Ac_mul_B!(y::AbstractVector, A::KroneckerMap, x::AbstractVector) = A_mul_B!(y, adjoint(A), x)
+
+###############
+# KroneckerSumMap
+###############
+struct KroneckerSumMap{T, As<:Tuple{LinearMap,LinearMap}} <: LinearMap{T}
+    maps::As
+    function KroneckerSumMap{T, As}(maps::As) where {T, As}
+        for A in maps
+            size(A, 1) == size(A, 2) || throw(ArgumentError("operators need to be square in Kronecker sums"))
+            promote_type(T, eltype(A)) == T || throw(InexactError())
+        end
+        return new{T,As}(maps)
+    end
+end
+
+KroneckerSumMap{T}(maps::As) where {T, As<:Tuple{LinearMap,LinearMap}} = KroneckerSumMap{T, As}(maps)
+
+"""
+    kronsum(A::LinearMap, B::LinearMap)
+    kronsum(A, B, Cs...)
+
+Construct a `KroneckerSumMap <: LinearMap` object, a (lazy) representation of the
+Kronecker sum `A⊕B = kron(A, Ib) + kron(Ia, B)` of two square `LinearMap`s. Here,
+`Ia` and `Ib` are identity operators of the size of `A` and `B`, respectively.
+Arguments of type `AbstractMatrix` are promoted to `LinearMap`s as long as there
+is a `LinearMap` object among the first 8 arguments.
+
+# Examples
+```jldoctest; setup=(using LinearAlgebra, SparseArrays, LinearMaps)
+julia> J = LinearMap(I, 2) # 2×2 identity map
+LinearMaps.UniformScalingMap{Bool}(true, 2)
+
+julia> E = spdiagm(-1 => trues(1)); D = LinearMap(E + E' - 2I);
+
+julia> Δ₁ = kron(D, J) + kron(J, D); # discrete 2D-Laplace operator
+
+julia> Δ₂ = LinearMaps.kronsum(D, D); # same operator as Kronecker sum
+
+julia> Matrix(Δ₁) == Matrix(Δ₂)
+true
+```
+"""
+kronsum(A::LinearMap{TA}, B::LinearMap{TB}) where {TA,TB} = KroneckerSumMap{promote_type(TA,TB)}((A, B))
+kronsum(A::LinearMap, B::LinearMap, C::LinearMap, Ds::LinearMap...) = kronsum(A, kronsum(B, C, Ds...))
+# promote AbstractMatrix arguments to LinearMaps, then take LinearMap-Kronecker sum
```

Comment:
As there is no `kronsum` in `Base`, this is probably not necessary. Though there is no problem in keeping it for consistency, and for the days when this function exists in Base :-)

Reply:
What do you mean? `KroneckerSum` currently expects `LinearMap` objects, or do you think we could relax this? The question is whether `mul!` continues to work for "mixed" `maps` tuples, or at least requires only mild modifications.

Reply:
My apologies for the late response. I meant that, since `kronsum` is not in base and does "belongs" to LinearMaps.jl, there is no type piracy in just defining
`kronsum(As::Vararg{Union{LinearMap,AbstractMatrix}})`

Reply:
I see, and I agree. I simplified the construction accordingly. I doubt that `kronsum` is gonna become a `Base` function. Without this structural view, where you keep the factors, it doesn't make much sense to add it.